src/OpenLoco/Windows/ToolbarTop.cpp

```diff
@@ -329,27 +331,28 @@ namespace OpenLoco::Ui::Windows::ToolbarTop::Game
     static void railroadMenuMouseDown(Window* window, WidgetIndex_t widgetIndex)
     {
         // Load dropdown objects removing any that are not unlocked.
-        registers regs;
-        regs.edi = X86Pointer(&available_objects[0]);
-        call(0x004A6841, regs);
-
-        // Sanity check: any objects available?
-        uint32_t i = 0;
-        while (available_objects[i] != 0xFF && i < std::size(available_objects))
-            i++;
-        if (i == 0)
+        // Note: This is not using player company id! This looks odd.
+        availableTracks = CompanyManager::get(GameCommands::getUpdatingCompanyId())->getAvailableRailTracks();
+
+        assert(std::size(available_objects) >= std::size(availableTracks));
+        // Legacy copy to available_objects remove when all users of 0x0050A006 accounted for
+        std::copy(std::begin(availableTracks), std::end(availableTracks), std::begin(available_objects));
+        available_objects[availableTracks.size()] = 0xFF;
+
+        if (availableTracks.size() == 0)
             return;
 
         auto company_colour = CompanyManager::getPlayerCompanyColour();
 
         // Add available objects to Dropdown.
         uint16_t highlighted_item = 0;
-        for (i = 0; available_objects[i] != 0xFF && i < std::size(available_objects); i++)
+        auto i = 0u;
```

Comment:
`i` can be declared as a loop variable in the `for` statement

Reply:
Its used to set the last dropdown item although i guess could just use std::size(availableTracks)